include/oneapi/dpl/internal/distributed_ranges_impl/shp/algorithms/copy.hpp

```diff
@@ -0,0 +1,229 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <memory>
+#include <type_traits>
+
+#include <sycl/sycl.hpp>
+
+#include <oneapi/dpl/internal/distributed_ranges_impl/concepts/concepts.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/detail/segments_tools.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/detail.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/device_ptr.hpp>
+#include <oneapi/dpl/internal/distributed_ranges_impl/shp/util.hpp>
+
+namespace oneapi::dpl::experimental::dr::shp
+{
+
+// Copy between contiguous ranges
+template <std::contiguous_iterator InputIt, std::contiguous_iterator OutputIt>
+requires __detail::is_syclmemcopyable<std::iter_value_t<InputIt>, std::iter_value_t<OutputIt>> sycl::event
+copy_async(InputIt first, InputIt last, OutputIt d_first)
+{
+    // auto &&q = __detail::default_queue();
```

Comment:
I guess these comments could be removed for better readability.

Reply:
ok, applied